driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -1768,8 +1773,6 @@ protected void initChannel(SocketChannel channel) throws Exception {
    * v5.
    */
   void switchToV5Framing() {
-    assert factory.protocolVersion.compareTo(ProtocolVersion.V5) >= 0;
```

Comment:
Why remove this?

Reply:
Hmm, I have no clue what I was thinking there. I'll revert.

Reply:
@adutra Now I remember why I removed this. It's legal for a client to not supply a specific version, e.g. not call `Cluster.Builder::withProtocolVersion`, in which case `factory.protocolVersion` is null. While v5 was marked beta this wasn't a problem as `Cluster.Builder::allowProtocolVersion` was required, but now this causes an NPE in the driver. In C*, `ProtocolNegotiationTest` has a test case that covers this.

Reply:
Thanks @beobal, I pushed a fix directly to 3.x, see 1af7382fe.